mesonbuild/dependencies/base.py

```diff
@@ -112,16 +115,6 @@ class DependencyMethods(Enum):
 
 class Dependency(HoldableObject):
 
-    @classmethod
-    def _process_include_type_kw(cls, kwargs: DependencyObjectKWs) -> str:
-        if 'include_type' not in kwargs:
-            return 'preserve'
-        if not isinstance(kwargs['include_type'], str):  # type: ignore[typeddict-item]
-            raise DependencyException('The include_type kwarg must be a string type')
-        if kwargs['include_type'] not in ['preserve', 'system', 'non-system']:  # type: ignore[typeddict-item]
-            raise DependencyException("include_type may only be one of ['preserve', 'system', 'non-system']")
-        return kwargs['include_type']  # type: ignore[typeddict-item]
```

Comment:
>  interpreter: port dependency include_type to typed_kargs
> 
> The cleanup this allows lower down points out that we don't properly validate the value passed to `as_system()`. I have no idea what happens if you pass a non-valid value, but it's a bug and I've simply made it a hard error. We can re-assess if necessary.

As far as I can tell this is totally fine, since `dependency('libzstd').as_system('hi')` triggers this precise DependencyException. Were you thinking of something else here?